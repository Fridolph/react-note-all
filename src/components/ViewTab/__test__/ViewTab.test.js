import React from 'react'
import {shallow} from 'enzyme'
import ViewTab from '../'

const props = {
  activeTab: 'list',
  onTabChange: jest.fn()
}

let wrapper

describe('@src/components/ViewTab', () => {
  beforeEach(() => {
    wrapper = shallow(<ViewTab {...props} />)
  })

  it('组件 snapshot 匹配', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('文字描述正确', () => {
    const listText = wrapper.find('.nav-item').first().find('span').text()
    expect(listText).toEqual('列表模式')

    const chartText = wrapper.find('.nav-item').last().find('span').text()
    expect(chartText).toEqual('图表模式')
  })

  it('能够正常进行点击事件', () => {
    const tabList = wrapper.find('.nav-item').first().find('a')
    tabList.simulate('click')
    expect(props.onTabChange).toHaveBeenCalledWith('list')

    const tabChart = wrapper.find('.nav-item').last().find('a')
    tabChart.simulate('click')
    expect(props.onTabChange).toHaveBeenCalledWith('chart')
  })
})
