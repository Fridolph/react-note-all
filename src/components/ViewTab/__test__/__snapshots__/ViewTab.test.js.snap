// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@src/components/ViewTab 组件 snapshot 匹配 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewTab
    activeTab="list"
    onTabChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="nav-item"
        >
          <a
            className="nav-link active"
            href="#"
            onClick={[Function]}
          >
            <Ionicon
              beat={false}
              className="rounded-circle mr-2"
              color="#007bff"
              fontSize="25px"
              icon="ios-paper"
              rotate={false}
              shake={false}
              style={Object {}}
            />
            <span>
              列表模式
            </span>
          </a>
        </li>,
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="#"
            onClick={[Function]}
          >
            <Ionicon
              beat={false}
              className="rounded-circle mr-2"
              color="#007bff"
              fontSize="25px"
              icon="ios-pie"
              rotate={false}
              shake={false}
              style={Object {}}
            />
            <span>
              图表模式
            </span>
          </a>
        </li>,
      ],
      "className": "nav nav-tabs nav-fill my-4",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="nav-link active"
            href="#"
            onClick={[Function]}
          >
            <Ionicon
              beat={false}
              className="rounded-circle mr-2"
              color="#007bff"
              fontSize="25px"
              icon="ios-paper"
              rotate={false}
              shake={false}
              style={Object {}}
            />
            <span>
              列表模式
            </span>
          </a>,
          "className": "nav-item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Ionicon
                beat={false}
                className="rounded-circle mr-2"
                color="#007bff"
                fontSize="25px"
                icon="ios-paper"
                rotate={false}
                shake={false}
                style={Object {}}
              />,
              <span>
                列表模式
              </span>,
            ],
            "className": "nav-link active",
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "beat": false,
                "className": "rounded-circle mr-2",
                "color": "#007bff",
                "fontSize": "25px",
                "icon": "ios-paper",
                "rotate": false,
                "shake": false,
                "style": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "列表模式",
              },
              "ref": null,
              "rendered": "列表模式",
              "type": "span",
            },
          ],
          "type": "a",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="nav-link"
            href="#"
            onClick={[Function]}
          >
            <Ionicon
              beat={false}
              className="rounded-circle mr-2"
              color="#007bff"
              fontSize="25px"
              icon="ios-pie"
              rotate={false}
              shake={false}
              style={Object {}}
            />
            <span>
              图表模式
            </span>
          </a>,
          "className": "nav-item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Ionicon
                beat={false}
                className="rounded-circle mr-2"
                color="#007bff"
                fontSize="25px"
                icon="ios-pie"
                rotate={false}
                shake={false}
                style={Object {}}
              />,
              <span>
                图表模式
              </span>,
            ],
            "className": "nav-link",
            "href": "#",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "beat": false,
                "className": "rounded-circle mr-2",
                "color": "#007bff",
                "fontSize": "25px",
                "icon": "ios-pie",
                "rotate": false,
                "shake": false,
                "style": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "图表模式",
              },
              "ref": null,
              "rendered": "图表模式",
              "type": "span",
            },
          ],
          "type": "a",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="nav-item"
          >
            <a
              className="nav-link active"
              href="#"
              onClick={[Function]}
            >
              <Ionicon
                beat={false}
                className="rounded-circle mr-2"
                color="#007bff"
                fontSize="25px"
                icon="ios-paper"
                rotate={false}
                shake={false}
                style={Object {}}
              />
              <span>
                列表模式
              </span>
            </a>
          </li>,
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="#"
              onClick={[Function]}
            >
              <Ionicon
                beat={false}
                className="rounded-circle mr-2"
                color="#007bff"
                fontSize="25px"
                icon="ios-pie"
                rotate={false}
                shake={false}
                style={Object {}}
              />
              <span>
                图表模式
              </span>
            </a>
          </li>,
        ],
        "className": "nav nav-tabs nav-fill my-4",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="nav-link active"
              href="#"
              onClick={[Function]}
            >
              <Ionicon
                beat={false}
                className="rounded-circle mr-2"
                color="#007bff"
                fontSize="25px"
                icon="ios-paper"
                rotate={false}
                shake={false}
                style={Object {}}
              />
              <span>
                列表模式
              </span>
            </a>,
            "className": "nav-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Ionicon
                  beat={false}
                  className="rounded-circle mr-2"
                  color="#007bff"
                  fontSize="25px"
                  icon="ios-paper"
                  rotate={false}
                  shake={false}
                  style={Object {}}
                />,
                <span>
                  列表模式
                </span>,
              ],
              "className": "nav-link active",
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "beat": false,
                  "className": "rounded-circle mr-2",
                  "color": "#007bff",
                  "fontSize": "25px",
                  "icon": "ios-paper",
                  "rotate": false,
                  "shake": false,
                  "style": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "列表模式",
                },
                "ref": null,
                "rendered": "列表模式",
                "type": "span",
              },
            ],
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="nav-link"
              href="#"
              onClick={[Function]}
            >
              <Ionicon
                beat={false}
                className="rounded-circle mr-2"
                color="#007bff"
                fontSize="25px"
                icon="ios-pie"
                rotate={false}
                shake={false}
                style={Object {}}
              />
              <span>
                图表模式
              </span>
            </a>,
            "className": "nav-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Ionicon
                  beat={false}
                  className="rounded-circle mr-2"
                  color="#007bff"
                  fontSize="25px"
                  icon="ios-pie"
                  rotate={false}
                  shake={false}
                  style={Object {}}
                />,
                <span>
                  图表模式
                </span>,
              ],
              "className": "nav-link",
              "href": "#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "beat": false,
                  "className": "rounded-circle mr-2",
                  "color": "#007bff",
                  "fontSize": "25px",
                  "icon": "ios-pie",
                  "rotate": false,
                  "shake": false,
                  "style": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "图表模式",
                },
                "ref": null,
                "rendered": "图表模式",
                "type": "span",
              },
            ],
            "type": "a",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
